### PR TITLE
Improve "is kernel compressed" check

### PR DESCRIPTION
**Pull request by cfergeau**
ARM64 kernels must be uncompressed before they can be booted.
vfkit has a check for this to warn the user before attempting to start the VM.
However this check has both false negatives and false positives (with an
uncompressed puipui linux kernel, and with a compressed fedora kernel)

This PR should fix both situations

### CODE REVIEW

File: pkg/vf/bootloader.go [hunk 1]
```diff
@@ -22,17 +42,7 @@ func isKernelUncompressed(filename string) (bool, error) {
...
-		return false, err
-	}
-	// uncompressed ARM64 kernels are matched as a MS executable, which is
-	// also an archive, so we need to special case it
-	if kind == matchers.TypeExe {
-		return true, nil
-	}
-
-	return false, nil
+	return isUncompressedArm64Kernel(buf), nil
```
Comment (praveenkumar) [inline pkg/vf/bootloader.go:45]: Where this `buf` variable comes from?

Comment (praveenkumar) [reply pkg/vf/bootloader.go:45]: nvm, got it.

**Comment by praveenkumar** [comment]
CI is failing as part of linter.

**Comment by cfergeau** [comment]
> CI is failing as part of linter.

Fixed now!

**Pull request merged**